In [29]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import matplotlib as mpl
mpl.rc('font',family='Malgun Gothic')
mpl.rcParams['axes.unicode_minus'] = False
import seaborn as sns
import scipy.stats as stats

In [11]:
df_ROS_Analysis = pd.read_csv('../../dataset/RecurrenceOfSurgeryAnalysis.csv')
df_ROS_Analysis[:5]

Unnamed: 0  입원기간  환자통증정도  통증기간(월)  Instability  수술기법  수술시간  수술실패여부  \
0           0     2      10      1.0            0  TELD  68.0       0   
1           1     1      10      1.0            0  TELD  31.0       0   
2           2     1       7      1.0            0  TELD  78.0       0   
3           3     1       7      2.0            0  TELD  73.0       0   
4           4     2       7      1.0            0  TELD  29.0       0   

    디스크단면적  MF + ES  고혈압여부  과거수술횟수  당뇨여부  우울증여부    체중   신장  성별  연령  헤모글로빈수치  \
0  2048.50   1824.6      0       0     0      0  60.3  163   2  66    13.66   
1  1753.10   1737.5      0       0     0      0  71.7  171   1  47    13.73   
2  1749.00   1188.5      0       1     0      0  77.1  178   1  39    15.34   
3  1796.20   2534.5      1       0     0      0  74.2  174   1  40    15.38   
4  2021.48   1840.6      1       0     0      0  80.7  183   1  42    13.08   

   Large Lymphocyte  
0              22.8  
1              44.9  
2              53.0  
3              54.7  
4              53.8

In [12]:
df_ROS = df_ROS_Analysis[['입원기간', '고혈압여부', '과거수술횟수', '당뇨여부', '우울증여부',
       '체중', '신장', '성별', '연령', '헤모글로빈수치', 'Large Lymphocyte']]
df_ROS

입원기간  고혈압여부  과거수술횟수  당뇨여부  우울증여부    체중   신장  성별  연령  헤모글로빈수치  \
0        2      0       0     0      0  60.3  163   2  66    13.66   
1        1      0       0     0      0  71.7  171   1  47    13.73   
2        1      0       1     0      0  77.1  178   1  39    15.34   
3        1      1       0     0      0  74.2  174   1  40    15.38   
4        2      1       0     0      0  80.7  183   1  42    13.08   
...    ...    ...     ...   ...    ...   ...  ...  ..  ..      ...   
1748    12      0       0     0      0  52.0  163   2  29    13.70   
1749     2      0       0     0      0  64.0  185   1  21    15.70   
1750     3      0       0     0      0  60.0  160   1  43    15.60   
1751     3      0       0     0      0  70.0  173   1  20    13.00   
1752     3      0       0     0      0  70.0  167   1  61    14.40   

      Large Lymphocyte  
0                 22.8  
1                 44.9  
2                 53.0  
3                 54.7  
4                 53.8  
...                ...  
1748              32.4  
1749              42.4  
1750              30.8  
1751               5.4  
1752              36.7  

[1753 rows x 11 columns]

In [13]:
df_ROS['체중_str'] = df_ROS['체중'].astype(str)
df_ROS['신장_str'] = df_ROS['신장'].astype(str)

C:\Users\01-02\AppData\Local\Temp\ipykernel_14332\2347698486.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_ROS['체중_str'] = df_ROS['체중'].astype(str)
C:\Users\01-02\AppData\Local\Temp\ipykernel_14332\2347698486.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_ROS['신장_str'] = df_ROS['신장'].astype(str)


In [14]:
df_ROS[['체중_str', '신장_str']]

체중_str 신장_str
0      60.3    163
1      71.7    171
2      77.1    178
3      74.2    174
4      80.7    183
...     ...    ...
1748   52.0    163
1749   64.0    185
1750   60.0    160
1751   70.0    173
1752   70.0    167

[1753 rows x 2 columns]

In [15]:
df_ROS['체중_str_num'] = pd.to_numeric(df_ROS['체중_str'])
df_ROS['체중_str_num']

C:\Users\01-02\AppData\Local\Temp\ipykernel_14332\2272525233.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_ROS['체중_str_num'] = pd.to_numeric(df_ROS['체중_str'])


0       60.3
1       71.7
2       77.1
3       74.2
4       80.7
        ... 
1748    52.0
1749    64.0
1750    60.0
1751    70.0
1752    70.0
Name: 체중_str_num, Length: 1753, dtype: float64

In [16]:
df_ROS['신장_str_num'] = pd.to_numeric(df_ROS['신장_str'])
df_ROS['신장_str_num'] 

C:\Users\01-02\AppData\Local\Temp\ipykernel_14332\628115756.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_ROS['신장_str_num'] = pd.to_numeric(df_ROS['신장_str'])


0       163
1       171
2       178
3       174
4       183
       ... 
1748    163
1749    185
1750    160
1751    173
1752    167
Name: 신장_str_num, Length: 1753, dtype: int64

In [17]:
df_ROS['신장_str_num_M'] = df_ROS['신장_str_num'] * 0.01
df_ROS['신장_str_num_M']

C:\Users\01-02\AppData\Local\Temp\ipykernel_14332\4168967935.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_ROS['신장_str_num_M'] = df_ROS['신장_str_num'] * 0.01


0       1.63
1       1.71
2       1.78
3       1.74
4       1.83
        ... 
1748    1.63
1749    1.85
1750    1.60
1751    1.73
1752    1.67
Name: 신장_str_num_M, Length: 1753, dtype: float64

In [18]:
df_ROS['BMI'] = df_ROS['체중_str_num']/(df_ROS['신장_str_num_M'] * df_ROS['신장_str_num_M'])
df_ROS['BMI']

C:\Users\01-02\AppData\Local\Temp\ipykernel_14332\607245305.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_ROS['BMI'] = df_ROS['체중_str_num']/(df_ROS['신장_str_num_M'] * df_ROS['신장_str_num_M'])


0       22.695623
1       24.520365
2       24.334049
3       24.507861
4       24.097465
          ...    
1748    19.571681
1749    18.699781
1750    23.437500
1751    23.388687
1752    25.099502
Name: BMI, Length: 1753, dtype: float64

In [19]:
df_ROS

입원기간  고혈압여부  과거수술횟수  당뇨여부  우울증여부    체중   신장  성별  연령  헤모글로빈수치  \
0        2      0       0     0      0  60.3  163   2  66    13.66   
1        1      0       0     0      0  71.7  171   1  47    13.73   
2        1      0       1     0      0  77.1  178   1  39    15.34   
3        1      1       0     0      0  74.2  174   1  40    15.38   
4        2      1       0     0      0  80.7  183   1  42    13.08   
...    ...    ...     ...   ...    ...   ...  ...  ..  ..      ...   
1748    12      0       0     0      0  52.0  163   2  29    13.70   
1749     2      0       0     0      0  64.0  185   1  21    15.70   
1750     3      0       0     0      0  60.0  160   1  43    15.60   
1751     3      0       0     0      0  70.0  173   1  20    13.00   
1752     3      0       0     0      0  70.0  167   1  61    14.40   

      Large Lymphocyte 체중_str 신장_str  체중_str_num  신장_str_num  신장_str_num_M  \
0                 22.8   60.3    163        60.3         163          1.63   
1                 44.9   71.7    171        71.7         171          1.71   
2                 53.0   77.1    178        77.1         178          1.78   
3                 54.7   74.2    174        74.2         174          1.74   
4                 53.8   80.7    183        80.7         183          1.83   
...                ...    ...    ...         ...         ...           ...   
1748              32.4   52.0    163        52.0         163          1.63   
1749              42.4   64.0    185        64.0         185          1.85   
1750              30.8   60.0    160        60.0         160          1.60   
1751               5.4   70.0    173        70.0         173          1.73   
1752              36.7   70.0    167        70.0         167          1.67   

            BMI  
0     22.695623  
1     24.520365  
2     24.334049  
3     24.507861  
4     24.097465  
...         ...  
1748  19.571681  
1749  18.699781  
1750  23.437500  
1751  23.388687  
1752  25.099502  

[1753 rows x 17 columns]

In [23]:
# 출처 Chat GPT
def classify_bmi_range(bmi): 
    if bmi < 18.5:
        return '저체중'
    elif 18.5 <= bmi < 23:
        return '정상'
    elif 23 <= bmi < 25:
        return '과체중'
    else:
        return '비만'

In [24]:
df_ROS['비만도'] = df_ROS['BMI'].apply(classify_bmi_range)
df_ROS

C:\Users\01-02\AppData\Local\Temp\ipykernel_14332\2319910938.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_ROS['비만도'] = df_ROS['BMI'].apply(classify_bmi_range)


입원기간  고혈압여부  과거수술횟수  당뇨여부  우울증여부    체중   신장  성별  연령  헤모글로빈수치  \
0        2      0       0     0      0  60.3  163   2  66    13.66   
1        1      0       0     0      0  71.7  171   1  47    13.73   
2        1      0       1     0      0  77.1  178   1  39    15.34   
3        1      1       0     0      0  74.2  174   1  40    15.38   
4        2      1       0     0      0  80.7  183   1  42    13.08   
...    ...    ...     ...   ...    ...   ...  ...  ..  ..      ...   
1748    12      0       0     0      0  52.0  163   2  29    13.70   
1749     2      0       0     0      0  64.0  185   1  21    15.70   
1750     3      0       0     0      0  60.0  160   1  43    15.60   
1751     3      0       0     0      0  70.0  173   1  20    13.00   
1752     3      0       0     0      0  70.0  167   1  61    14.40   

      Large Lymphocyte 체중_str 신장_str  체중_str_num  신장_str_num  신장_str_num_M  \
0                 22.8   60.3    163        60.3         163          1.63   
1                 44.9   71.7    171        71.7         171          1.71   
2                 53.0   77.1    178        77.1         178          1.78   
3                 54.7   74.2    174        74.2         174          1.74   
4                 53.8   80.7    183        80.7         183          1.83   
...                ...    ...    ...         ...         ...           ...   
1748              32.4   52.0    163        52.0         163          1.63   
1749              42.4   64.0    185        64.0         185          1.85   
1750              30.8   60.0    160        60.0         160          1.60   
1751               5.4   70.0    173        70.0         173          1.73   
1752              36.7   70.0    167        70.0         167          1.67   

            BMI  비만도  
0     22.695623   정상  
1     24.520365  과체중  
2     24.334049  과체중  
3     24.507861  과체중  
4     24.097465  과체중  
...         ...  ...  
1748  19.571681   정상  
1749  18.699781   정상  
1750  23.437500  과체중  
1751  23.388687  과체중  
1752  25.099502   비만  

[1753 rows x 18 columns]

In [22]:
df_ROS[['입원기간', '고혈압여부', '과거수술횟수', '당뇨여부', '우울증여부',
        '성별', '연령', '헤모글로빈수치', 'Large Lymphocyte', 'BMI', '비만도']]

입원기간  고혈압여부  과거수술횟수  당뇨여부  우울증여부  성별  연령  헤모글로빈수치  Large Lymphocyte  \
0        2      0       0     0      0   2  66    13.66              22.8   
1        1      0       0     0      0   1  47    13.73              44.9   
2        1      0       1     0      0   1  39    15.34              53.0   
3        1      1       0     0      0   1  40    15.38              54.7   
4        2      1       0     0      0   1  42    13.08              53.8   
...    ...    ...     ...   ...    ...  ..  ..      ...               ...   
1748    12      0       0     0      0   2  29    13.70              32.4   
1749     2      0       0     0      0   1  21    15.70              42.4   
1750     3      0       0     0      0   1  43    15.60              30.8   
1751     3      0       0     0      0   1  20    13.00               5.4   
1752     3      0       0     0      0   1  61    14.40              36.7   

            BMI  비만도  
0     22.695623   정상  
1     24.520365  과체중  
2     24.334049  과체중  
3     24.507861  과체중  
4     24.097465  과체중  
...         ...  ...  
1748  19.571681   정상  
1749  18.699781   정상  
1750  23.437500  과체중  
1751  23.388687  과체중  
1752  25.099502   비만  

[1753 rows x 11 columns]

In [ ]:
df_ROS.info()

### 연속형/연속형
- 목표변수 : '입원기간'
- 설정변수 : '연령', '헤모글로빈수치', 'Large Lymphocyte'

#### 선정 변수 입원기간, 연령
- 분석 내용 : 연령에 따른 입원기간에 변화가 있는가?
    - 귀무가설 : 변화가 없다.
    - 대립가설 : 변화가 있다.
- 분석 결론 (기준 : p-value 0.05)
    - 통계 결론 : 귀무가설 참. (pvalue=0.0772836)
    - 사실 결론 : 변화가 없다.

In [25]:
df_ROS_age = df_ROS[['입원기간', '연령']]
df_ROS_age[:3]

입원기간  연령
0     2  66
1     1  47
2     1  39

In [26]:
df_ROS_age.isnull().sum()

입원기간    0
연령      0
dtype: int64

In [27]:
df_ROS_age.corr()

입원기간        연령
입원기간  1.000000  0.042207
연령    0.042207  1.000000

In [30]:
stats.pearsonr(df_ROS_age['연령'],df_ROS_age['입원기간'])

PearsonRResult(statistic=0.04220662718944215, pvalue=0.07728361514743073)

#### 선정 변수 입원기간, 헤모글로빈수치
- 분석 내용 : 헤모글로빈수치에 따른 입원기간에 변화가 있는가?
    - 귀무가설 : 변화가 없다.
    - 대립가설 : 변화가 있다.
- 분석 결론 (기준 : p-value 0.05)
    - 통계 결론 : 귀무가설 참. (pvalue=0.1597244)
    - 사실 결론 : 변화가 없다.

In [31]:
df_ROS_hemo = df_ROS[['입원기간', '헤모글로빈수치']]
df_ROS_hemo[:3]

입원기간  헤모글로빈수치
0     2    13.66
1     1    13.73
2     1    15.34

In [32]:
df_ROS_hemo.isnull().sum()

입원기간       0
헤모글로빈수치    0
dtype: int64

In [33]:
df_ROS_hemo.corr()

입원기간   헤모글로빈수치
입원기간     1.000000 -0.033596
헤모글로빈수치 -0.033596  1.000000

In [35]:
stats.pearsonr(df_ROS_hemo['헤모글로빈수치'],df_ROS_hemo['입원기간'])

PearsonRResult(statistic=-0.03359551158366978, pvalue=0.159724499158464)

#### 선정 변수 입원기간, Large Lymphocyte
- 분석 내용 : Large Lymphocyte에 따른 입원기간에 변화가 있는가?
    - 귀무가설 : 변화가 없다.
    - 대립가설 : 변화가 있다. 
- 분석 결론 (기준 : p-value 0.05)
    - 통계 결론 : 귀무가설 참. (pvalue=0.134780)
    - 사실 결론 : 변화가 없다.

In [36]:
df_ROS_LL = df_ROS[['입원기간', 'Large Lymphocyte']]
df_ROS_LL[:3]

입원기간  Large Lymphocyte
0     2              22.8
1     1              44.9
2     1              53.0

In [38]:
df_ROS_LL.isnull().sum()

입원기간                0
Large Lymphocyte    0
dtype: int64

In [39]:
df_ROS_LL.corr()

입원기간  Large Lymphocyte
입원기간              1.000000         -0.035733
Large Lymphocyte -0.035733          1.000000

In [42]:
stats.pearsonr(df_ROS_LL['Large Lymphocyte'],df_ROS_LL['입원기간'])

PearsonRResult(statistic=-0.035733081571023506, pvalue=0.13478055798954264)

### 범주형/연속형
- 목표변수 : '입원기간'
- 설정변수 : '고혈압여부', '과거수술횟수', '당뇨여부', '우울증여부', '성별', '비만도'

#### 선정 변수 입원기간, 고혈압여부
- 분석 내용 : 고혈압여부에 따라 입원기간이 달라지는가?
    - 귀무가설 : 고혈압여부에 따라 입원기간이 달라지지 않는다.
    - 대립가설 : 고혈압여부에 따라 입원기간이 달라진다.
- 분석 결론 (기준 : p-value 0.05)
    - 통계 결론 : 대립가설 참. (pvalue=2.8377648684295895e-133)
    - 사실 결론 : 고혈압여부에 따라 입원기간이 달라진다.

In [43]:
df_ROS_press = df_ROS[['입원기간', '고혈압여부']]
df_ROS_press[:3]

입원기간  고혈압여부
0     2      0
1     1      0
2     1      0

In [48]:
df_ROS_press['고혈압여부'].value_counts()

0    1521
1     232
Name: 고혈압여부, dtype: int64

In [49]:
df_pressure = df_ROS_press[['고혈압여부', '입원기간']]
df_pressure0 = df_pressure[df_pressure['고혈압여부'] == 0]
df_pressure0[:2]

고혈압여부  입원기간
0      0     2
1      0     1

In [51]:
df_pressure1 = df_pressure[df_pressure['고혈압여부'] == 1]
df_pressure1[:2]

고혈압여부  입원기간
3      1     1
4      1     2

In [52]:
stats.ranksums(df_pressure0['고혈압여부'], df_pressure1['고혈압여부'])

RanksumsResult(statistic=-24.567130472603523, pvalue=2.8377648684295895e-133)

#### 선정 변수 입원기간, 과거수술횟수
- 분석 내용 : 과거수술횟수에 따라 입원기간이 달라지는가?
    - 귀무가설 : 달라지지 않는다.
    - 대립가설 : 달라진다.
- 분석 결론 (기준 : p-value 0.05)
    - 통계 결론 : 대립가설 참. (pvalue=0.0)
    - 사실 결론 : 과거수술횟수에 따라 입원기간이 달라진다.

In [55]:
df_ROS[['과거수술횟수']].value_counts()

과거수술횟수
0         1378
1          215
2          152
3            8
dtype: int64

In [57]:
df_past = df_ROS[['과거수술횟수', '입원기간']]

In [58]:
past_list = df_past['과거수술횟수'].value_counts().index.to_list()
past_list

[0, 1, 2, 3]

In [59]:
df_past_sur = [] 
for item in past_list : 
    condition = (df_past['과거수술횟수'] == item)
    df_past_sur.append(df_past[condition])

In [60]:
len(df_past_sur)

4

In [62]:
stats.kruskal(df_past_sur[0]['과거수술횟수'], df_past_sur[1]['과거수술횟수'], df_past_sur[2]['과거수술횟수'], df_past_sur[3]['과거수술횟수'])

KruskalResult(statistic=1752.0000000000011, pvalue=0.0)

#### 선정 변수 입원기간, 당뇨여부
- 분석 내용 : 당뇨여부에 따른 입원기간이 달라지는가?
    - 귀무가설 : 달라지지 않는다.
    - 대립가설 : 달라진다.
- 분석 결론 (기준 : p-value 0.05)
    - 통계 결론 : 대립가설 참. (pvalue=4.738463770660181e-68)
    - 사실 결론 : 당뇨여부에 따른 입원기간이 달라진다.

In [63]:
df_ROS_sugar = df_ROS[['입원기간', '당뇨여부']]
df_ROS_sugar[:3]

입원기간  당뇨여부
0     2     0
1     1     0
2     1     0

In [64]:
df_ROS_sugar['당뇨여부'].value_counts()

0    1645
1     108
Name: 당뇨여부, dtype: int64

In [65]:
df_sugar = df_ROS_sugar[['당뇨여부', '입원기간']]
df_sugar0 = df_sugar[df_sugar['당뇨여부'] == 0]
df_sugar0[:2]

당뇨여부  입원기간
0     0     2
1     0     1

In [67]:
df_sugar1 = df_sugar[df_sugar['당뇨여부'] == 1]
df_sugar1[:2]

당뇨여부  입원기간
6      1     1
75     1     1

In [68]:
stats.ranksums(df_sugar0['당뇨여부'], df_sugar1['당뇨여부'])

RanksumsResult(statistic=-17.43173687270112, pvalue=4.738463770660181e-68)

#### 선정 변수 입원기간, 성별
- 분석 내용 : 성별에 따라 입원기간이 달라지는가?
    - 귀무가설 : 달라지지 않는다.
    - 대립가설 : 달라진다.
- 분석 결론 (기준 : p-value 0.05)
    - 통계 결론 : 대립가설 참. (pvalue=8.982823573911684e-271)
    - 사실 결론 : 성별에 따라 입원기간이 달라진다.

In [69]:
df_ROS_sex = df_ROS[['입원기간', '성별']]
df_ROS_sex[:3]

입원기간  성별
0     2   2
1     1   1
2     1   1

In [71]:
df_ROS_sex['성별'].value_counts()

1    1090
2     663
Name: 성별, dtype: int64

In [72]:
df_sex = df_ROS_sex[['성별', '입원기간']]
df_sex1 = df_sex[df_sex['성별'] == 1]
df_sex1[:2]

성별  입원기간
1   1     1
2   1     1

In [73]:
df_sex2 = df_sex[df_sex['성별'] == 2]
df_sex2[:2]

성별  입원기간
0   2     2
5   2     1

In [76]:
stats.ranksums(df_sex1['성별'], df_sex2['성별'])

RanksumsResult(statistic=-35.157326807910486, pvalue=8.982823573911684e-271)

#### 선정 변수 입원기간, 우울증여부
- 분석 내용 : 우울증여부에 따라 입원기간이 달라지는가?
    - 귀무가설 : 달라지지 않는다.
    - 대립가설 : 달라진다.
- 분석 결론 (기준 : p-value 0.05)
    - 통계 결론 : 대립가설 참. (pvalue=0.0)
    - 사실 결론 : 우울증여부에 따라 입원기간이 달라진다.

In [78]:
df_ROS[['우울증여부']].value_counts()

우울증여부
0        1710
1          42
2           1
dtype: int64

In [79]:
df_sad = df_ROS[['우울증여부', '입원기간']]

In [80]:
sad_list = df_sad['우울증여부'].value_counts().index.to_list()
sad_list

[0, 1, 2]

In [81]:
df_sad_sure = [] 
for item in sad_list : 
    condition = (df_sad['우울증여부'] == item)
    df_sad_sure.append(df_sad[condition])

In [82]:
stats.kruskal(df_sad_sure[0]['우울증여부'], df_sad_sure[1]['우울증여부'],  df_sad_sure[2]['우울증여부'])

KruskalResult(statistic=1752.000000000005, pvalue=0.0)

#### 선정 변수 입원기간, 비만도
- 분석 내용 : 비만도에 따라 입원기간이 달라지는가?
    - 귀무가설 : 달라지지 않는다.
    - 대립가설 : 달라진다.
- 분석 결론 (기준 : p-value 0.05)
    - 통계 결론 : 대립가설 참. (pvalue=0.0)
    - 사실 결론 : 비만도에 따라 입원기간이 달라진다.

In [83]:
df_ROS[['비만도']].value_counts()

비만도
과체중    779
정상     519
비만     421
저체중     34
dtype: int64

In [84]:
df_obesity = df_ROS[['비만도', '입원기간']]

In [93]:
obesity_list = df_obesity['비만도'].value_counts().index.to_list()
obesity_list

['과체중', '정상', '비만', '저체중']

In [89]:
df_obesity_sure = [] 
for item in obesity_list : 
    condition = (df_obesity['비만도'] == item)
    df_obesity_sure.append(df_obesity[condition])

In [94]:
stats.kruskal(df_obesity_sure[0]['비만도'], df_obesity_sure[1]['비만도'], df_obesity_sure[2]['비만도'],  df_obesity_sure[3]['비만도'])

KruskalResult(statistic=1752.0000000000002, pvalue=0.0)

### 분석 스토리 
- 연령에 따른 입원기간에 변화가 있는지 pearsonr 방식으로 검증을 해보았다. 
결과적으로 pvalue가 0.0772836이 나왔으므로 귀무가설에 의해 변화가 없는 것으로 나왔다.
따라서 입원기간은 연령과 무관한것으로 보인다.

- 림프구에 따른 입원기간에 변화를 pearsonr 방식으로 검증을 해보았다. 
결과적으로 pvalue가 0.134780이 나왔으므로 귀무가설에 의해 변화가 없는 것으로 나왔다.
따라서 면역체계를 담당하는 림프구와 입원기간은 무관한 것으로 보인다.

- 당뇨여부에 따른 입원기간의 변화를 ranksums 방식으로 검증을 해보았다.
결과적으로 pvalue가 4.738..... - 68이 나왔으므로 대립가설에 의해 변화가 있는 것으로 나왔다.
따라서 당뇨여부는 입원기간에 영향을 주는 것으로 보인다.

- 과거수술횟수에 따른 입원기간의 변화를 kruskal 방식으로 검증을 해보았다.
결과적으로 pvalue가 0으로 나왔기 때문에 대립가설이 참인것으로 나왔다.
따라서 과거수술횟수는 입원기간에 영향을 주는 것으로 보인다.

- 비만도에 따른 입원기간의 변화를 kruskal 방식으로 검증을 해보았다.
결과적으로 pvalue가 0이므로 대립가설에 의해 변화가 있는 것으로 나왔다.
따라서 비만도는 입원기간에 영향을 주는 것으로 보인다.